.github/workflows/benchmark.yml

```
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Benchmarks
+on:
+  push:
+    branches: [main]
+    paths:
+      - ".github/workflows/benchmark.yml"
+      - "ci/scripts/bench_adapt.py"
```

If we have `paths` for `push`, this workflow isn't executed for most commits on main. Because most commits don't change these files.

How about using `paths` only for `pull_request`?  

```suggestion
  push:
    branches: [main]
  pull_request:
    paths:
      - ".github/workflows/benchmark.yml"
      - "ci/scripts/bench_adapt.py"
```

If we have `pull_request.paths`, we can test this workflow only in PRs that change these files such as this PR.

---

Fair enough, Thank you so much :+1: